processing/src/main/java/io/druid/query/spec/LegacySegmentSpec.java

```diff
@@ -49,17 +49,10 @@
       throw new IAE("Unknown type[%s] for intervals[%s]", intervals.getClass(), intervals);
     }
 
-    return Lists.transform(
-        intervalStringList,
-        new Function<Object, Interval>()
-        {
-          @Override
-          public Interval apply(Object input)
-          {
-            return new Interval(input);
-          }
-        }
-    );
+    return intervalStringList
+        .stream()
+        .map(input -> new Interval(input, ISOChronology.getInstanceUTC()))
+        .collect(Collectors.toList());
```

Comment:
This is functionally different. The prior method had wrappers on fetching the underlying values, this materializes a new copy of the list, but it should be ok for here since you are discarding the prior list anyways. I don't think this is called in any tight loops, so should be safe to cause a bit of extra heap garbage.